### ISSUE TITLE
Ref does not work on custom components created with applyTheme

### ISSUE THREAD

**Issue by MrToph** [bumbag/bumbag-ui#71]
React refs do not seem to work correctly on custom components created with `applyTheme`.
This seems to be because they are a functional component and they don't correctly forwardRef?

I get this error:

```
Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
```

**To Reproduce**

```typescript
const CustomButton = applyTheme(Button, {
  styles: {
    base: {
      minHeight: "unset",
      whiteSpace: `nowrap`,
    },
  },
  defaultProps: {
    variant: "ghost",
  },
});

const App = () => {
  const buttonRef = useRef(null);
  const handleClick = () => {
    console.log(buttonRef.current) // should be defined
  }
  return (
    <CustomButton onClick={handleClick} ref={buttonRef}>
      Click
    </CustomButton>
  );
};

```

**Comment by jxom** [comment]
Thanks! I'll have a look into this soon! Maybe use `elementRef` for now?

**Comment by MrToph** [comment]
Great. I'm not aware of `elementRef`, I checked the docs of the `Button` component but couldn't find anything. Can you point me to some description of `elementRef`

**Comment by jxom** [comment]
This should be fixed on `1.3.8`
